logpuzzle.py

```diff
@@ -12,24 +12,47 @@
 Given an apache logfile, find the puzzle urls and download the images.
 
 Here's what a puzzle url looks like:
-10.254.254.28 - - [06/Aug/2007:00:13:48 -0700] "GET /~foo/puzzle-bar-aaab.jpg HTTP/1.0" 302 528 "-" "Mozilla/5.0 (Windows; U; Windows NT 5.1; en-US; rv:1.8.1.6) Gecko/20070725 Firefox/2.0.0.6"
+10.254.254.28 - - [06/Aug/2007:00:13:48 -0700]
+"GET /~foo/puzzle-bar-aaab.jpg HTTP/1.0" 302 528 "-"
+"Mozilla/5.0 (Windows; U; Windows NT 5.1; en-US; rv:1.8.1.6)
+Gecko/20070725 Firefox/2.0.0.6"
 
 """
+___author___ = "Justin Miller internet resources noted thru code"
 
 import os
 import re
 import sys
-import urllib
+import urllib.request
 import argparse
 
 
+def part_c_sort(match):
+    """Returns 2nd set of letters in filename p-1111-2222.jpg"""
+    match = match.split('-')[-1]
+    return match
+
+
 def read_urls(filename):
     """Returns a list of the puzzle urls from the given log file,
     extracting the hostname from the filename itself.
     Screens out duplicate urls and returns the urls sorted into
     increasing order."""
     # +++your code here+++
-    pass
+    with open(filename, 'r') as rf:
+        text = rf.read()
+    matches = re.findall(r'GET (\S*puzzle\S*) HTTP', text)
+    # https://www.geeksforgeeks.org/python-ways-to-remove-duplicates-from-list/
+    matches = set(matches)
+    if filename.startswith('place'):
+        # https://www.w3schools.com/python/ref_list_sort.asp
+        matches = sorted(matches, key=part_c_sort)
+    else:
+        matches = sorted(matches)
+    full_paths = []
+    for match in matches:
+        full_paths.append('http://code.google.com' + match)
```

Comment:
The base url (code.google.com) should be derived from the functions input `filename` rather than hard-coded